beluga/include/beluga/actions/reweight.hpp

```diff
@@ -0,0 +1,113 @@
+   * sensor updates.
+   */
+  template <
+      class ExecutionPolicy,
+      class Range,
+      class Model,
+      std::enable_if_t<std::is_execution_policy_v<std::decay_t<ExecutionPolicy>>, int> = 0,
+      std::enable_if_t<ranges::range<Range>, int> = 0>
+  constexpr auto operator()(ExecutionPolicy&& policy, Range& range, Model model) const -> Range& {
+    static_assert(beluga::is_particle_range_v<Range>);
+    auto states = range | beluga::views::states | ranges::views::common;
+    auto weights = range | beluga::views::weights | ranges::views::common;
+    std::transform(
+        policy,               //
+        std::begin(states),   //
+        std::end(states),     //
+        std::begin(weights),  //
+        std::begin(weights),  //
+        [model = std::move(model)](const auto& s, auto w) { return w * model(s); });
+    return range;
+  }
+
+  /// Overload that re-orders arguments from a view closure.
+  template <
+      class Range,
+      class Model,
+      class ExecutionPolicy,
+      std::enable_if_t<ranges::range<Range>, int> = 0,
+      std::enable_if_t<std::is_execution_policy_v<ExecutionPolicy>, int> = 0>
+  constexpr auto operator()(Range&& range, Model model, ExecutionPolicy policy) const -> Range& {
+    return (*this)(std::move(policy), std::forward<Range>(range), std::move(model));
+  }
+
+  /// Overload that returns a view closure to compose with other views.
+  template <class ExecutionPolicy, class Model, std::enable_if_t<std::is_execution_policy_v<ExecutionPolicy>, int> = 0>
+  constexpr auto operator()(ExecutionPolicy policy, Model model) const {
+    return ranges::make_action_closure(ranges::bind_back(reweight_base_fn{}, std::move(model), std::move(policy)));
+  }
+};
+
+/// Implementation detail for a reweight range adaptor object with a default execution policy.
+struct reweight_fn : public reweight_base_fn {
+  using reweight_base_fn::operator();
+
+  /// Overload that defines a default execution policy.
+  template <class Range, class Model, std::enable_if_t<ranges::range<Range>, int> = 0>
+  constexpr auto operator()(Range&& range, Model model) const -> Range& {
+    return (*this)(std::execution::seq, std::forward<Range>(range), std::move(model));
+  }
+
+  /// Overload that returns a view closure to compose with other views.
+  template <class Model>
+  constexpr auto operator()(Model model) const {
+    return ranges::make_action_closure(ranges::bind_back(reweight_fn{}, std::move(model)));
+  }
+};
+
+}  // namespace detail
+
+/// [Range adaptor object](https://en.cppreference.com/w/cpp/named_req/RangeAdaptorObject) that
+/// can update the weights in a particle range using a sensor model.
+/**
+ * This step computes the importance factor or weight of each particle to incorporate measurements.
+ * The importance is proportional to the probability of seeing the measurement given the current particle state.
```

Comment:
@nahueespinosa nit: perhaps consider:
```suggestion
 * This action updates particle weights by importance weight multiplication. 
 * These importance weights are computed by a given measurement likelihood 
 * function (or sensor model) for current particle states.
```